File: RepoTransBench/target_projects/Python/Java/HuobiRDCenter_huobi_Python/src/test/java/com/huobi/public_tests/TestPublicUtilsCoverage.java
package com.huobi.public_tests;

import org.junit.jupiter.api.Test;
import static org.junit.jupiter.api.Assertions.*;

import java.util.*;

public class TestPublicUtilsCoverage {

    // Simulated input_checker public version
    public static class InputChecker {
        public static Object checkShouldNotNone(Object val, String name) {
            if (val == null) {
                throw new IllegalArgumentException(name + " should not be null");
            }
            return null;
        }
        public static Object checkShouldNone(Object val, String name) {
            if (val != null) {
                throw new IllegalArgumentException(name + " should be null");
            }
            return null;
        }
    }

    // Simulated UrlParamsBuilder public variant
    public static class UrlParamsBuilder {
        private final LinkedHashMap<String, String> params = new LinkedHashMap<>();
        public UrlParamsBuilder putUrl(String k, String v) {
            params.put(k, v);
            return this;
        }
        public String buildUrl() {
            if (params.isEmpty()) return "";
            StringBuilder sb = new StringBuilder("?");
            for (Map.Entry<String, String> entry : params.entrySet()) {
                if (sb.length() > 1) sb.append("&");
                sb.append(entry.getKey()).append("=").append(entry.getValue());
            }
            return sb.toString();
        }
    }

    // Simulated logging class
    public static class LogInfo {
        public static void printWarn(String message) {}
        public static void printBasicInfo(String message) {}
        public static void printReplace(String from, String to) {}
    }

    // Simulated time_service
    public static class TimeService {
        public static int getCurrentTimestamp() {
            return (int) (System.currentTimeMillis() / 1000);
        }
    }

    // Simulated print_mix_object
    public static class PrintMixObject {
        public static void printBasicObject(Object obj) {}
        public static void printList(List<?> l) {}
        public static void printDict(Map<String, ?> d) {}
        public static void printBasicDict(Map<String, ?> d) {}
        public static void printBasicList(List<?> l) {}
    }

    // Simulated json_parser
    public static class JsonParser {
        public static String jsonDumps(Map<String, Object> map) {
            StringBuilder sb = new StringBuilder("{");
            for (Map.Entry<String, Object> entry : map.entrySet()) {
                if (sb.length() > 1) sb.append(",");
                sb.append("\"").append(entry.getKey()).append("\":\"").append(entry.getValue().toString()).append("\"");
            }
            sb.append("}");
            return sb.toString();
        }
        public static Map<String, String> jsonLoads(String s) {
            Map<String, String> result = new HashMap<>();
            s = s.trim();
            s = s.substring(1, s.length()-1);
            String[] kvs = s.split(",");
            for (String kv : kvs) {
                if (kv.trim().length() == 0) continue;
                String[] parts = kv.split("\":\"");
                if (parts.length == 2) {
                    String key = parts[0].replaceAll("^\"", "");
                    String val = parts[1].replaceAll("\"$", "");
                    result.put(key, val);
                }
            }
            return result;
        }
    }

    @Test
    public void testCheckShouldNotNone() {
        assertThrows(IllegalArgumentException.class, () ->
            InputChecker.checkShouldNotNone(null, "different_param"));
    }

    @Test
    public void testCheckShouldNotNoneValid() {
        assertDoesNotThrow(() ->
            InputChecker.checkShouldNotNone(123, "another_param"));
    }

    @Test
    public void testCheckShouldNone() {
        assertDoesNotThrow(() -> InputChecker.checkShouldNone(null, "wonka_param"));
        assertThrows(IllegalArgumentException.class, () -> InputChecker.checkShouldNone(0, "wonka_param"));
    }

    @Test
    public void testAddAndBuildUrl() {
        UrlParamsBuilder builder = new UrlParamsBuilder();
        builder.putUrl("x", "alpha");
        builder.putUrl("y", "beta");
        String url = builder.buildUrl();
        assertTrue(url.equals("?x=alpha&y=beta") || url.equals("?y=beta&x=alpha"));
        builder = new UrlParamsBuilder();
        assertEquals("", builder.buildUrl());
    }

    @Test
    public void testLogMethodsExist() {
        LogInfo.printWarn("public warn message");
        LogInfo.printBasicInfo("public info message");
        LogInfo.printReplace("public_from", "public_to");
    }

    @Test
    public void testGetCurrentTime() {
        int now = TimeService.getCurrentTimestamp();
        assertTrue(now >= 0);
    }

    public static class Dummy {
        @Override
        public String toString() { return "OtherDummy"; }
    }

    @Test
    public void testPrintObjectBasic() {
        Dummy obj = new Dummy();
        PrintMixObject.printBasicObject(obj);
    }

    @Test
    public void testPrintListAndDict() {
        Dummy obj = new Dummy();
        PrintMixObject.printList(Arrays.asList(obj, obj));
        PrintMixObject.printList(Arrays.asList(obj));
        PrintMixObject.printDict(Collections.singletonMap("b", 2));
        PrintMixObject.printDict(Collections.singletonMap("a", 42));
        PrintMixObject.printBasicDict(Collections.singletonMap("z", 789));
        PrintMixObject.printBasicList(Arrays.asList(9,8,7));
        PrintMixObject.printBasicList(Arrays.asList(0));
    }

    @Test
    public void testJsonParse() {
        Map<String, Object> obj = new HashMap<>();
        obj.put("spam", "eggs");
        String jsonStr = JsonParser.jsonDumps(obj);
        Map<String, String> result = JsonParser.jsonLoads(jsonStr);
        assertEquals("eggs", result.get("spam"));
    }
}

File: RepoTransBench/target_projects/Python/Rust/HuobiRDCenter_huobi_Python/tests/test_public_utils_coverage.rs
// test_public_utils_coverage.rs

#[cfg(test)]
mod tests {
    use std::collections::HashMap;

    // Simulate input_checker functionality (with different params than original)
    mod input_checker {
        pub fn check_should_not_none<T>(val: Option<T>, _param: &str) -> Result<(), String> {
            match val {
                Some(_) => Ok(()),
                None => Err("Value is None".to_string()),
            }
        }
        pub fn check_should_none<T>(val: Option<T>, _param: &str) -> Result<(), String> {
            match val {
                None => Ok(()),
                Some(_) => Err("Value is not None".to_string()),
            }
        }
    }

    mod url_params_builder {
        use std::collections::HashMap;
        #[derive(Default)]
        pub struct UrlParamsBuilder {
            pub params: HashMap<String, String>,
        }
        impl UrlParamsBuilder {
            pub fn new() -> Self {
                Self { params: HashMap::new() }
            }
            pub fn put_url(&mut self, k: &str, v: &str) {
                self.params.insert(k.to_string(), v.to_string());
            }
            pub fn build_url(&self) -> String {
                if self.params.is_empty() {
                    "".to_string()
                } else {
                    let mut pairs: Vec<String> = self.params.iter().map(|(k, v)| format!("{}={}", k, v)).collect();
                    pairs.sort();
                    format!("?{}", pairs.join("&"))
                }
            }
        }
    }
    mod log_info {
        pub fn print_warn(_msg: &str) {}
        pub fn print_basic_info(_msg: &str) {}
        pub fn print_replace(_from: &str, _to: &str) {}
    }
    mod time_service {
        pub fn get_current_timestamp() -> i64 {
            7654321 // just a different value
        }
    }
    mod print_mix_object {
        pub fn print_basic_object<T: std::fmt::Debug>(obj: &T) {
            let _ = format!("{:?}", obj);
        }
        pub fn print_list<T: std::fmt::Debug>(lst: &[T]) {
            for obj in lst {
                let _ = format!("{:?}", obj);
            }
        }
        pub fn print_dict<V: std::fmt::Debug>(dict: &std::collections::HashMap<&str, V>) {
            for (k, v) in dict {
                let _ = format!("{}:{:?}", k, v);
            }
        }
        pub fn print_basic_dict<V: std::fmt::Debug>(dict: &std::collections::HashMap<&str, V>) {
            for (k, v) in dict {
                let _ = format!("{}:{:?}", k, v);
            }
        }
        pub fn print_basic_list<T: std::fmt::Debug>(lst: &[T]) {
            for v in lst {
                let _ = format!("{:?}", v);
            }
        }
    }
    mod json_parser {
        use serde_json::{json, Value};
        pub fn json_dumps(v: &serde_json::Value) -> String {
            serde_json::to_string(v).unwrap()
        }
        pub fn json_loads(s: &str) -> serde_json::Value {
            serde_json::from_str(s).unwrap()
        }
    }

    #[test]
    fn test_check_should_not_none() {
        assert!(input_checker::check_should_not_none::<i32>(None, "different_param").is_err());
    }

    #[test]
    fn test_check_should_not_none_valid() {
        assert!(input_checker::check_should_not_none(Some(123), "another_param").is_ok());
    }

    #[test]
    fn test_check_should_none() {
        assert!(input_checker::check_should_none::<i32>(None, "wonka_param").is_ok());
        // Edge: check_should_none(0, ...) => error
        assert!(input_checker::check_should_none(Some(0), "wonka_param").is_err());
    }

    #[test]
    fn test_add_and_build_url() {
        let mut builder = url_params_builder::UrlParamsBuilder::new();
        builder.put_url("x", "alpha");
        builder.put_url("y", "beta");
        let url = builder.build_url();
        let valid1 = "?x=alpha&y=beta";
        let valid2 = "?y=beta&x=alpha";
        assert!(url == valid1 || url == valid2, "url was {}", url);
        let builder2 = url_params_builder::UrlParamsBuilder::new();
        assert_eq!(builder2.build_url(), "");
    }

    #[test]
    fn test_log_methods_exist() {
        log_info::print_warn("public warn message");
        log_info::print_basic_info("public info message");
        log_info::print_replace("public_from", "public_to");
    }

    #[test]
    fn test_get_current_time() {
        let now = time_service::get_current_timestamp();
        assert!(now >= 0);
    }

    struct Dummy;
    impl std::fmt::Debug for Dummy {
        fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result { write!(f, "OtherDummy") }
    }
    #[test]
    fn test_print_object_basic() {
        let obj = Dummy;
        print_mix_object::print_basic_object(&obj);
    }

    #[test]
    fn test_print_list_and_dict() {
        let obj = Dummy;
        print_mix_object::print_list(&[obj, obj]);
        print_mix_object::print_list(&[obj]);
        let mut dict = std::collections::HashMap::new();
        dict.insert("b", 2);
        dict.insert("a", 42);
        print_mix_object::print_dict(&dict);
        let mut basicdict = std::collections::HashMap::new();
        basicdict.insert("z", 789);
        print_mix_object::print_basic_dict(&basicdict);
        print_mix_object::print_basic_list(&[9, 8, 7]);
        print_mix_object::print_basic_list(&[0]);
    }

    #[test]
    fn test_parse() {
        use serde_json::json;
        let obj = json!({"spam": "eggs"});
        let json_str = json_parser::json_dumps(&obj);
        let result = json_parser::json_loads(&json_str);
        assert_eq!(result["spam"], "eggs");
    }
}